src/core/mac/mac.cpp

```diff
@@ -910,26 +912,32 @@ void Mac::HandleBeginTransmit(Timer &aTimer)
     aTimer.GetOwner<Mac>().HandleBeginTransmit();
 }
 
-void Mac::HandleBeginTransmit(void)
+Frame *Mac::GetOperationFrame(void)
 {
-    Frame *theFrame = NULL;
-    otError error = OT_ERROR_NONE;
-    bool applyTransmitSecurity = true;
+    Frame *frame = NULL;
 
     switch (mOperation)
     {
     case kOperationTransmitOutOfBandFrame:
-        theFrame = mOobFrame;
+        frame = mOobFrame;
         break;
 
     default:
-        theFrame = mTxFrame;
+        frame = mTxFrame;
         break;
     }
 
-    assert(theFrame != NULL);
+    assert(frame != NULL);
+
+    return frame;
+}
 
-    Frame &sendFrame(*theFrame);
+void Mac::HandleBeginTransmit(void)
+{
+    otError error = OT_ERROR_NONE;
+    bool applyTransmitSecurity = true;
+    Frame *opFrame = GetOperationFrame();
+    Frame &sendFrame(*opFrame);
```

Comment:
How about if we remove `otFrame` variable and just initialize the `sendFrame` here directly by calling the method?